pkg/apis/extensions/v1beta1/register.go

```diff
@@ -60,6 +58,7 @@ func addKnownTypes() {
 		&ListOptions{},
 		&ConfigMap{},
 		&ConfigMapList{},
+		&v1.DeleteOptions{},
```

Comment:
> Because clients need a consistent view of the world.  Different groups
> evolve at different times - when we lock and create a version we need
> to identify the recognized options.

That sounds a lot like an argument saying that no other API group should embed a `v1.PodSpec`.  Given that we know we want to do exactly that, I'm not completely convinced.